scripts/src/javatests/org/oppia/android/scripts/license/LicenseTextsCheckTest.kt

```diff
@@ -0,0 +1,119 @@
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("values")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_noArgsPassed_failWithException() {
+    val thirdPartyDepsXmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    thirdPartyDepsXmlFile.writeText(WARNING_COMMENT)
+
+    val exception = assertThrows(Exception::class) {
+      main(arrayOf())
+    }
+
+    assertThat(exception).hasMessageThat().contains(TOO_FEW_ARGS_PASSED_FAILURE)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_emptyXmlFile_checkFailsWithException() {
+    val thirdPartyDepsXmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    thirdPartyDepsXmlFile.writeText("")
+
+    val exception = assertThrows(Exception::class) {
+      main(arrayOf("${tempFolder.root}/values/third_party_dependencies.xml"))
+    }
+
+    assertThat(exception).hasMessageThat().contains(LICENSE_TEXTS_CHECKED_IN_FAILURE)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_warningCommentNotPresent_checkFailsWithException() {
+    val thirdPartyDepsXmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    thirdPartyDepsXmlFile.writeText(
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <resources>
+        <string name="third_party_dependency_name_0">Glide</string>
+      </resources>
+      """.trimIndent()
+    )
+
+    val exception = assertThrows(Exception::class) {
+      main(arrayOf("${tempFolder.root}/values/third_party_dependencies.xml"))
+    }
+
+    assertThat(exception).hasMessageThat().contains(LICENSE_TEXTS_CHECKED_IN_FAILURE)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_onlyWarningCommentPresent_checkPasses() {
+    val thirdPartyDepsXmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    thirdPartyDepsXmlFile.writeText(WARNING_COMMENT)
+
+    main(arrayOf("${tempFolder.root}/values/third_party_dependencies.xml"))
+
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_MESSAGE)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_warningCommentPresentWithSomeXmlCode_checkPasses() {
+    val thirdPartyDepsXmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    thirdPartyDepsXmlFile.writeText(
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      $WARNING_COMMENT
+      <resources>
+        <string name="third_party_dependency_name_0">Glide</string>
+      </resources>
+      """.trimIndent()
+    )
+
+    main(arrayOf("${tempFolder.root}/values/third_party_dependencies.xml"))
+
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_MESSAGE)
+  }
+
+  @Test
+  fun testLicenseTexsCheck_xmlFileNotPresent_checkFailsWithFileNotFoundException() {
+    val exception = assertThrows(FileNotFoundException::class) {
+      main(arrayOf("${tempFolder.root}/values/third_party_dependencies.xml"))
+    }
+
+    assertThat(exception).hasMessageThat().contains("No such file or directory")
```

Comment:
Nit: we should actually throw a custom error exception (e.g. with ``check(file.exists()) { "File does not exist: $file" }``) since the exact "No such file or directory" can change between JVM implementations.

Reply:
Done.